src/main/java/net/earthcomputer/clientcommands/command/AliasCommand.java

```diff
@@ -0,0 +1,174 @@
+        String cmd;
+        if(aliasMap.containsKey(aliasKey)) {
+            cmd = aliasMap.get(aliasKey);
+        } else {
+            sendError(new LiteralText("No such key exists"));
+            return 0;
+        }
+
+        if(arguments!=null) {
+            cmd = String.format(cmd, arguments.split(" "));
+        }
+        MinecraftClient.getInstance().player.sendChatMessage(cmd);
+
+        return 0;
+    }
+
+    private static int addAlias(String key, String command) {
+        ClientPlayNetworkHandler networkHandler = MinecraftClient.getInstance().getNetworkHandler();
+        assert networkHandler != null;
+        var dispatcher = (CommandDispatcher<ServerCommandSource>) (CommandDispatcher<?>) networkHandler.getCommandDispatcher();
+
+        HashMap<String, String> aliasMap = getAliases();
+
+        assert aliasMap != null;
+        if(aliasMap.containsKey(key)) {
+            sendError(new LiteralText("That key already exists!"));
+            return 0;
+        }
+        addClientSideCommand(key);
+
+        dispatcher.register(literal(key)
+                .executes(ctx -> executeAliasCommand(key, null))
+                .then(argument("arguments", greedyString())
+                        .executes(ctx -> executeAliasCommand(key, getString(ctx, "arguments")))));
+        aliasMap.put(key, command);
+
+        try {
+            Writer writer = new FileWriter(String.valueOf(aliasPath));
+            Gson gson = new Gson();
+
+            gson.toJson(aliasMap, writer);
+            writer.flush();
+            writer.close();
+
+            sendFeedback(new LiteralText("New alias added successfully!"));
+        } catch (Exception e) {
+            sendError(new LiteralText("Uh oh"));
+        }
+        return 0;
+    }
+    private static int listAliases() {
+        HashMap<String, String> aliasMap = getAliases();
+        if (aliasMap == null || aliasMap.isEmpty()) {
+            sendError(new LiteralText("No aliases registered"));
+        } else {
+            sendFeedback(new LiteralText("Aliases registered: " + aliasMap.size()));
+
+            for(String key: aliasMap.keySet()) {
+                sendFeedback(Formatting.BOLD + key + Formatting.RESET+ ": "+ aliasMap.get(key).replace("%","%%"));
+            }
+
+        }
+        return 0;
+    }
+    private static int removeAlias(String key) {
+        ClientPlayNetworkHandler networkHandler = MinecraftClient.getInstance().getNetworkHandler();
+        assert networkHandler != null;
+        var dispatcher = (CommandDispatcher<ServerCommandSource>) (CommandDispatcher<?>) networkHandler.getCommandDispatcher();
+
+        HashMap<String, String> aliasMap = getAliases();
+
+        if(aliasMap != null && aliasMap.containsKey(key)) {
+            BrigadierRemover.of(dispatcher).get(key).remove();
+            aliasMap.remove(key);
+        } else {
+            sendError(new LiteralText("No such key exists"));
```

Comment:
Again here, I think this is user feedback, so no exception should be thrown.

Reply:
If it’s actually feedback, I think `sendFeedback` should be used. I always try to avoid `sendError`, and throw an exception instead.
This style is also documented here: [CONTRIBUTING.md](https://github.com/Earthcomputer/clientcommands/blob/CONTRIBUTING.md).
The only time I used `sendError` was when I couldn’t possibly throw an exception (`cplayerinfo`).